batch/plugins/org.jboss.tools.batch.core/src/org/jboss/tools/batch/core/BatchCorePlugin.java

```diff
@@ -28,6 +34,23 @@ public static BatchCorePlugin getDefault() {
 		return plugin;
 	}
 
+	@Override
+	public void start(BundleContext context) throws Exception {
```

Comment:
Save participants are stored in org.eclipse.core.internal.resources.SaveManager in Map<String, ISaveParticipant> so that there can be no more than one listener for a plugin.

We have 4 plugins that add participant by the same pattern with anonymous class: WebKbPlugin, SeamCorePlugin, CDICorePlugin, BatchCorePlugin. Yes, it may be better to release reference on stop.